pallets/lbp/src/lib.rs

```diff
@@ -1157,6 +1153,11 @@ impl<T: Config> AMM<T::AccountId, AssetId, AssetPair, BalanceOf<T>> for Pallet<T
 				Error::<T>::MaxInRatioExceeded
 			);
 
+			ensure!(
```

Comment:
I think we should also add negative test cases triggering the InsufficientAssetBalance error, both when we buy accumulated and distributed assets.